examples/devices/amplifier.py

```diff
@@ -89,7 +88,7 @@ async def set_amplification(self, amplification: float) -> None:
         self.device.amplification = amplification
 
 
-@dataclass
+@pydantic_dataclass
```

Comment:
I would favour `BaseModel` over pydantic dataclass unless there is a specific reason to go with it. In my opinion, dataclasses are a bit shoehorned in and the maintainers focus on supporting `BaseModel`.

Reply:
Dataclasses allow our current positional constructor usage, while BaseModels do not (even if the dataclasses do complain in the IDE, BaseModels throw an exception at runtime).
Currently handling for discriminated unions is not a drop in replacement when using BaseModels
Prior to change classes were dataclasses, now they are pydantic dataclasses. Smaller change than dataclass -> BaseModel. Might be worth doing when moving to Pydantic2 but for now I am pushing back on logic of "if there's not anything obviously wrong merge it" ;)

Extending BaseModel would allow us to not have to repeat the annotation on new subclasses and gives us more convenient .dict() [.model_dump() in Pydantic2], .json [.json_dump()], rather than ignoring type checking that dataclass.asdict shouldn't work on Pydantic dataclasses.

Reply:
Cleared with CF, he says OK